flang/lib/Optimizer/Builder/IntrinsicCall.cpp

```diff
@@ -5745,6 +5745,14 @@ IntrinsicLibrary::genReduce(mlir::Type resultType,
   int rank = arrayTmp.rank();
   assert(rank >= 1);
 
+  // Arguements to the reduction operation are passed by reference or value?
+  bool argByRef = true;
+  if (auto embox =
+          mlir::dyn_cast_or_null<fir::EmboxProcOp>(operation.getDefiningOp())) {
```

Comment:
Does REDUCE works with dummy procedure and procedure pointers? If so it would be good to add tests for those cases to ensure the pattern matching here works with them.

Reply:
> Does REDUCE works with dummy procedure and procedure pointers? If so it would be good to add tests for those cases to ensure the pattern matching here works with them.

I'll check if this is supported and add proper test if it is.

Reply:
So it works but there is I need to check for the dummy procedure by value because the information is lost in FIR. I'll prepare a follow up patch and we can discuss on it.